extensions/ql-vscode/src/databases/database-fetcher.ts

```diff
@@ -294,16 +296,18 @@ export async function importArchiveDatabase(
   cli: CodeQLCliServer,
 ): Promise<DatabaseItem | undefined> {
   try {
-    const item = await databaseArchiveFetcher(
+    const origin: DatabaseOrigin = {
+      type: databaseUrl.endsWith(".testproj") ? "testproj" : "archive",
+      // TODO validate that archive origins can use a file path, not a URI
```

Comment:
I don't think we currently use the origin for archives anywhere, so it should be fine to change this to an `fsPath`. However, if we do start using this in the future, we do need to handle both URIs and file paths for archives.

Reply:
If in the future we do want to use URIs, I'd suggest adding a new field `uri` to hold this value. We've had bugs in the past where we assigned URI strings to `path` properties.